torch/_dynamo/variables/builder.py

```diff
@@ -1512,6 +1513,17 @@ def update_dim2constraint(dim, constraint_range):
 
     return dynamic_dims, constraint_dims
 
+def replace_leaf_with_fake_tensor_and_record(e, tx, ignore_subclass=False, *, source: Optional[Source], is_tensor: bool):
+    if torch._is_functional_tensor(e):
+        from torch._C._functorch import _wrap_functional_tensor, _unwrap_functional_tensor, current_level
+
+        reapply_views = torch._C._functionalization_reapply_views_tls()
+        unwrap_e = _unwrap_functional_tensor(e, reapply_views)
+        replaced_e = replace_leaf_with_fake_tensor_and_record(unwrap_e, tx, ignore_subclass, source=source, is_tensor=is_tensor)
+        assert torch._is_functional_tensor(replaced_e) and isinstance(replaced_e, FakeTensor)
+        return replaced_e
```

Comment:
Uh I'm confused. I would expect that we do the following:
- FunctionalTensor(to_fake(unwrap(e))). I do not see where we are constructing a new FunctionalTensor.

Reply:
I think it's because we didn't disable the functionalize key, so in wrap_to_fake_tensor_and_record, it eventually calls empty_strided gives us a FunctionalTensor. It further wraps the FunctionalTensor with FakeTensor.

Is there any difference between FakeTensor(FunctionalTensor) vs FunctionalTensor(FakeTensor)? I mean from dispatcher's point of view, they should be the same?